### ISSUE TITLE
[SmartPower] Timberborn 0.5.4.2 (experimental) crashing while loading a save

### ISSUE THREAD

**Issue by danielclasen** [ihsoft/TimberbornMods#20]
Hi @ihsoft 

there seems to be another issue related to #18 or your fix in b67e256. This NullReferenceException is thrown while loading a save with a ton of Engines/PowerGenerators already built:

```
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
Timberborn.PowerGenerating.GoodPoweredGenerator.UpdateGoodConsumption () (at <9c3d4d3201d44f47984b37c4d361d709>:0)
Timberborn.PowerGenerating.GoodPoweredGenerator.PostInitializeLoadedEntity () (at <9c3d4d3201d44f47984b37c4d361d709>:0)
Timberborn.Persistence.WorldEntitiesLoader.PostInitializeEntity (Timberborn.Persistence.InstantiatedEntitySave entitySave) (at <407eebbe57a04fd0a437a0e7b9913473>:0)
Timberborn.Persistence.WorldEntitiesLoader.PostInitializeEntities (System.Collections.Generic.IEnumerable`1[T] entitySaves) (at <407eebbe57a04fd0a437a0e7b9913473>:0)
Timberborn.Persistence.WorldEntitiesLoader.PostLoadNonSingletons () (at <407eebbe57a04fd0a437a0e7b9913473>:0)
Timberborn.SingletonSystem.SingletonLifecycleService.PostLoadNonSingletons () (at <6e8a35ad066f451bb3cd3d1cd0ec7b12>:0)
(wrapper dynamic-method) Timberborn.SingletonSystem.SingletonLifecycleService.DMD<Timberborn.SingletonSystem.SingletonLifecycleService::LoadAll>(Timberborn.SingletonSystem.SingletonLifecycleService)
Timberborn.SingletonSystem.SingletonLifecycleAdapter.Start () (at <6e8a35ad066f451bb3cd3d1cd0ec7b12>:0)

[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
Timberborn.CoreUI.PanelStack.Awake () (at <3121d35c35e241dcaad0b87ff11a064c>:0)
```

With those dependencies installed:

```
[Message:   BepInEx] BepInEx 5.4.22.0 - Timberborn (04.12.2023 17:22:09)
[Info   :   BepInEx] Running under Unity v2023.1.20.3515684
[Info   :   BepInEx] CLR runtime version: 4.0.30319.42000
[Info   :   BepInEx] Supports SRE: True
[Info   :   BepInEx] System platform: Bits64, Windows
[Message:   BepInEx] Preloader started
[Info   :   BepInEx] Loaded 1 patcher method from [BepInEx.Preloader 5.4.22.0]
[Info   :   BepInEx] 1 patcher plugin loaded
[Info   :   BepInEx] Patching [UnityEngine.CoreModule] with [BepInEx.Chainloader]
[Message:   BepInEx] Preloader finished
[Message:   BepInEx] Chainloader ready
[Message:   BepInEx] Chainloader started
[Info   :   BepInEx] 2 plugins to load
[Info   :   BepInEx] Loading [Mod Manager UI 1.3.6]
[Info   :   BepInEx] Loading [TimberAPI BepInEx Entrypoint 1.0.0]
[Message:   BepInEx] Chainloader startup complete
[Info   : Unity Log] Starting game version 0.5.4.2-37f7207-xsw
...
[Info   : Unity Log] Modded: true, BepInEx
[Info   :TimberAPI Core] TimberAPI 0.6.3.1 - Timberborn 0.5.4.2, loader: BepInEx
[Info   : TimberAPI] Mod loading started
[Info   : TimberAPI] Found 12 mods
[Info   : TimberAPI] Deserialized 12 mods
[Info   :Extended Architecture] Loaded version 0.4.0.0
[Info   :FerrisWheel] Loaded version 1.1.0.0
[Info   :    Ladder] Loaded version 1.2.7.0
[Info   :More Platforms] Loaded version 1.3.2.0
[Info   :Vertical Power Shaft] Loaded version 1.3.0.0
[Info   :Storage1x1x2] Loaded version 2.1.5.0
[Info   :Extended Floodgate] Loaded version 0.8.4.0
[Info   : Flywheels] Loaded Flywheels Entrypoint
[Info   : Flywheels] Loaded version 1.0.0.0
[Info   :Path Extention] Loaded version 0.1.6.0
[Info   :SmartPower] Loaded version 1.5.0.0
[Info   :FloodgateTriggersPlugin] FloodgateTriggersPlugin is loaded.
[Info   :FloodgateTriggersPlugin] Loaded version 4.0.4.0
[Info   : Staircase] Loaded version 1.5.5.0
[Info   : TimberAPI] Mod loading finished in: 1192ms, Loaded mods:12
```

Would you be so kind to have another look into the error?

**Comment by danielclasen** [comment]
Just figured out that I had some unfinished engines in that save file. So I disabled the mod, restarted the game, loaded the save file, finished the constructions and created a new save. Now everything is working as expected with the mod enabled again.

**Comment by ihsoft** [comment]
Here is a hotfix:
[SmartPower_hotfix1.zip](https://github.com/ihsoft/TimberbornMods/files/13620529/SmartPower_hotfix1.zip)

Full fix and a new version will be out later. Maybe today evening, or maybe on weekend.

**Comment by danielclasen** [comment]
You're a gentleman and a scholar! Thank you :)

**Comment by ihsoft** [comment]
Fixed in `1.6`
